# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/editing/crashtests/justifycenter-then-delete-selection-on-DOMSubtreeModified.html
# Commit: f3218564df77
# Full Hash: f3218564df7730e438567383fbbf84b3fa25f91d
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-12-07 04:11:18
# Description:
#   Bug 1803626 - Make `HTMLEditor::AlignNodesAndDescendants` check whether insertion point stays valid r=m_kato
#   
#   In the test case, `deleteFromDocument` call makes the point unset.  We should
#   make it check whether a legacy mutation event listener creates unexpected result
#   after calling `RemoveAlignFromDescendants`.
# ==============================================================================

diff -r 0f9cd4dbd0d1 -r f3218564df77 testing/web-platform/tests/editing/crashtests/justifycenter-then-delete-selection-on-DOMSubtreeModified.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/editing/crashtests/justifycenter-then-delete-selection-on-DOMSubtreeModified.html	Tue Dec 06 07:33:14 2022 +0000
@@ -0,0 +1,24 @@
+<!doctype html>
+<html>
+<head>
+<meta charset="utf-8">
+<script>
+document.addEventListener("DOMContentLoaded", () => {
+  document.execCommand("selectAll");
+  document.querySelector("li").addEventListener("DOMSubtreeModified", () => {
+    document.designMode = "on";
+    document.execCommand("justifyCenter");
+    getSelection().deleteFromDocument();
+  });
+  document.querySelector("li").type = "1";
+});
+</script>
+</head>
+<body>
+<li>
+  <h3 align="right">
+    <option contenteditable="true"></option>
+  </h3>
+</li>
+</body>
+</html>
